### ISSUE TITLE
Since vector - effect options other than non - scaling - stroke already have multiple implementations, is the "at risk" flag valid?

### ISSUE THREAD

**Issue by satakagi** [w3c/svgwg#306]
Gecko patch: https://bugzilla.mozilla.org/show_bug.cgi?id=1318208
Polyfill: http://svg2.mbsrv.net/devinfo/devstd/non-scaling-objects-2/VectorEffect_polyfill_wScreen.html

**Comment by jarek-foksa** [comment]
AFAIK `non-scaling-stroke` and "none" are the only values that have any implementations at all, thus the "at risk" flag seems to be justified.

**Comment by nikosandronikos** [comment]
FWIW we were told that a polyfill is a valid implementation.

**Comment by svgeesus** [comment]
Yes, a polyfill is a valid implementation for the purpose of of CR exit criteria.

**Comment by tabatkins** [comment]
Tho, for web tech, polyfills aren't useful guarantors of quality/stability without additional assurances from browsers.

**Comment by JonFerraiolo** [comment]
My opinion: drop everything but non scaling stroke. Lighten the burden on user agents, authoring tools, converters, book authors, students, and the WG itself. Keep it simple. In fact, I would not cry if the entire feature were dropped. If this set of functionality can't be achieved easily in Javascript, then better to add better DOM APIs into the spec and/or write a how-to blog post than add automagic features used only a tiny fraction of language users. 

Note that I have had a requirement for a non scaling stroke with SVG1 and found that I could achieve the desired result if I organized my transformations in the right way. It was not difficult.

**Comment by satakagi** [comment]
@Tavmjong 
This is a test involving polyfill for that.
http://svgmap.org/devinfo/devstd/non-scaling-objects-2/VectorEffect_polyfill_noTf.html

However, I think that this does not conform to the requirements of WPT, so it is necessary to repair?

**Comment by satakagi** [comment]
Made pull req. for tests 
https://github.com/web-platform-tests/wpt/pull/13302

**Comment by dirkschulze** [comment]
We have implementations in Firefox, WebKit and Blink (Latter 2 counting as one implementation as implementation happened before fork.) That should fulfill requirements already but didn’t test Edge yet.

**Comment by dirkschulze** [comment]
We have the other property values in the SVG 2.1 implementation. I suggest moving the other properties out of SVG 2.0.

**Comment by longsonr** [comment]
The Firefox patches have not landed and have not been properly reviewed. They might require significant rework before landing and they might not even work or be mergable any more. Are you sure you want to count them as one of your implementatations?

> We have implementations in Firefox, WebKit and Blink (Latter 2 counting as one implementation as implementation happened before fork.) That should fulfill requirements already but didn’t test Edge yet.

The Firefox bug contains code that has not landed in any branch, it may no longer apply and it certainly hasn't been reviewed or tested. That surely doesn't count as an implementation does it?

**Comment by dirkschulze** [comment]
@longsonr No, it doesn't. Was my misunderstanding then. Are there plans to review/land it?

**Comment by longsonr** [comment]
There are no plans to review it or land it without commitment from other vendors.

**Comment by Tavmjong** [comment]
@satakagi I've implemented the rest of the vector effects in Inkscape (less than 20 lines of code!) but I'm not sure about one thing. In your polyfill, checking the 'non-scaling-size' or the 'non-rotation' box causes the dotted shapes to move to the corresponding solid shape. Is this really correct? I would not expect the translations to be removed on the dotted shapes when one of these vector effect values is used.

**Comment by satakagi** [comment]
@dirkschulze Please tell me the criteria of counting as an implementation. The implementation on the forked gecko was made certainly as in the above patch, but it has not landed on the main stream. As discussed in bugzilla, it will not enter the mainstream unless other vendors adopt it.

**Comment by dirkschulze** [comment]
@satakagi There must be 2 independent implementations. The implementation must be accessible and testable. The SVG WG has the additional requirement that at least one of the 2 implementations must be in one of the major browser (Firefox, Safari, Chrome, Edge) and must at least ship in a preview version with the feature accessible by anyone. (Could be disabled by default but must be able to be enabled for testing.)

**Comment by satakagi** [comment]
@dirkschulze In the context of svgwg, the forked gecko implementation does not satisfy the requirements of svgwg. By the way, where is the requirement stated?
Well, I remember that this specification was created based on the task of listing requirements for SVG 2 when most browser vendors participated, but unfortunately the current browser vendor's The situation seems to have changed.
As you can see from the discussion in bugzilla, this seems to be falling into a chicken or egg dilemma.
This situation does not seem to change as long as multiple implementations with polyfill or authoring tool do not meet the requirements.
At least we feel that contributing by native implementation to browsers is no longer difficult.

**Comment by dirkschulze** [comment]
@satakagi Lets discuss the details in the next WG call. I'll find previous resolutions for that meeting.

**Comment by dirkschulze** [comment]
We heard positive feedback from Chrome and WebKit giving the same statement as @longsonr. If other implementations implement they'll follow. Let's wait a little bit longer and we may not need to decide if a published patch counts as implementation.
